src/Button/Button.component.tsx

```diff
@@ -460,10 +467,12 @@ export const Button = ({
     circular,
     children,
     minWidth,
-    icon: Icon,
+    prefix,
+    suffix,
     ...props
 }: ButtonProps): React.ReactElement<ButtonProps> => {
-    const iconOnly = Icon && React.Children.count(children) === 0;
+    const iconOnly =
+        (prefix ? !suffix : !!suffix) && React.Children.count(children) === 0;
```

Comment:
>`(prefix ? !suffix : !!suffix)`

This is effectively XOR

Reply:
Add a comment explaining what's going on here. This borders on too clever. :)